src/libraries/Microsoft.Extensions.Logging.Configuration/src/Microsoft.Extensions.Logging.Configuration.csproj

```
@@ -36,4 +36,8 @@
     <ProjectReference Include="$(LibrariesProjectRoot)Microsoft.Extensions.Options.ConfigurationExtensions\src\Microsoft.Extensions.Options.ConfigurationExtensions.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <ProjectReference Include="$(LibrariesProjectRoot)Microsoft.Extensions.Configuration.Binder\gen\Microsoft.Extensions.Configuration.Binder.SourceGeneration.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="true" />
```

><ProjectReference Include="$(LibrariesProjectRoot)Microsoft.Extensions.Configuration.Binder\gen\Microsoft.Extensions.Configuration.Binder.SourceGeneration.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="true" /> [](http://example.com/codeflow?start=4&length=222)

does this mean we'll have the source gen packaged inside two packages now?

---

It's just meant to be consumed here, not redistributed.  When we redistribute a generator we'll use https://github.com/dotnet/runtime/blob/9d7c9d06f3e0a935ad7b68fee367e578b614c8aa/src/libraries/Microsoft.Extensions.Configuration.Binder/src/Microsoft.Extensions.Configuration.Binder.csproj#L35

@layomia can you double check that this doesn't make it into the nuspec as a dependency?  Just double check we have nothing mentioning Microsoft.Extensions.Configuration.Binder.SourceGeneration in the Microsoft.Extensions.Logging.Configuration package.

---

Actually, test failure shows this is a problem:
```
 error NU1101: (NETCORE_ENGINEERING_TELEMETRY=Build) Unable to find package Microsoft.Extensions.Configuration.Binder.SourceGeneration. No packages exist with this id
```

Fix this by applying `PrivateAssets="All"` to the project reference.
```suggestion
    <ProjectReference Include="$(LibrariesProjectRoot)Microsoft.Extensions.Configuration.Binder\gen\Microsoft.Extensions.Configuration.Binder.SourceGeneration.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="true" PrivateAssets="All" />
```